addons/website_sale_coupon/controllers/main.py

```diff
@@ -16,6 +17,27 @@ def pricelist(self, promo, **post):
             request.session['error_promo_code'] = coupon_status['error']
         return request.redirect(post.get('r', '/shop/cart'))
 
+    @http.route(['/shop/cart/update_json'], type='json', auth="public", methods=['POST'], website=True, csrf=False)
```

Comment:
```suggestion
    @http.route()
```
don't add param to overide routes